### PR TITLE
Feat cart management

### PR DESCRIPTION
**Pull request by aishwarya-bm**
add to cart,increment,decrement,delete implemented

https://user-images.githubusercontent.com/66695068/160280349-2a825c51-595b-4535-bb8e-95aaba40ca10.mp4

### CODE REVIEW

File: src/components/navpills/Navpills.jsx [hunk 1]
```diff
@@ -1,8 +1,16 @@
...
 export default function Navpills() {
-  //   const { state: cartState } = useCart();
+  const { stateCart, dispatchCart } = useCart();
+  const { dispatchUser } = useLogin();
   //   const { state: wishlistState } = useWishlist();
+  const signoutUser = () => {
+    localStorage.removeItem("userToken");
+    dispatchCart({ type: "RESET_CART" });
+    dispatchUser({ type: "LOGOUT_USER" });
+  };
```
Comment (ttahm3d) [inline src/components/navpills/Navpills.jsx]: This function can be moved to context. Destructure the call and add it in onClick

File: src/hooks/useAxios.jsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        setResponse(response);
+      } else if (method === "get") {
+        const response = await axios.get(url, { headers: header });
+        console.log("await", response);
+        setResponse(response);
+      }
+    } catch (err) {
+      console.log(err);
+    }
+  };
```
Comment (ttahm3d) [inline src/hooks/useAxios.jsx:20]: Please let me know if this works. 

i moved away from useAxios after I figured out that all my api calls are inside functions within context

File: src/utils/cart-checkout.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+              (Number(curr.orig_price) - Number(curr.curr_price)),
+          0
+        )
+
+const calculatePrice = (cart) =>  cart.reduce(
+          (acc, curr) => acc + Number(curr.qty) * Number(curr.orig_price),
+          0
+        )
+        
+export {calculateDiscount,calculatePrice}
```
Comment (ttahm3d) [inline src/utils/cart-checkout.js:14]: Nicely done. Same could have been done for cartSize too? or is there anything that I am missing here?

File: src/utils/cartitem-actions.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
+import axios from "axios";
+
+const addToCart = async (product, dispatchCart,navigate) => {
+  let response;
```
Comment (ttahm3d) [inline src/utils/cartitem-actions.js]: this can be a const and inside try block

**Review by ttahm3d — COMMENTED** [review]
Nice work on getting things done.

One suggestion that I would give is to re-export contexts using index files, that makes the code look clean. On the positive side, I fliked the way you have handled the functions that changes state separately.

File: src/components/cartprice/CartPrice.jsx [hunk 5]
```diff
@@ -9,22 +12,21 @@ export default function CartPrice() {
           <hr></hr>
```
Comment (kjais1720) [inline src/components/cartprice/CartPrice.jsx:12]: hr is a self closing tag, you can remove the closing tag here

File: src/components/navpills/Navpills.jsx [hunk 6]
```diff
@@ -1,8 +1,16 @@
 import { Link } from "react-router-dom";
+import { useCart } from "../../contexts/cart-context/cart-context";
+import { useLogin } from "../../contexts/login-context/login-context";
```
Comment (kjais1720) [inline src/components/navpills/Navpills.jsx]: you can import the files via index.js in context folder to make the import links shorter and readable

**Review by kjais1720 — APPROVED** [review]
Code looks clean and readable, just a couple of minor changes suggested

File: src/components/navpills/Navpills.jsx (hunk 1 above)
Comment (aishwarya-bm) [reply src/components/navpills/Navpills.jsx]: made separate login util for user signup/login functions

File: src/hooks/useAxios.jsx (hunk 2 above)
Comment (aishwarya-bm) [reply src/hooks/useAxios.jsx:20]: i tried this but did not seem to work, but kept it to revisit when time permits. will let you know how it goes

File: src/utils/cartitem-actions.js (hunk 4 above)
Comment (aishwarya-bm) [reply src/utils/cartitem-actions.js]: made changes

File: src/components/cartprice/CartPrice.jsx (hunk 5 above)
Comment (aishwarya-bm) [reply src/components/cartprice/CartPrice.jsx:12]: error is thrown asking to provide closing tag, even I had previously given in self closing manner

File: src/components/navpills/Navpills.jsx (hunk 6 above)
Comment (aishwarya-bm) [reply src/components/navpills/Navpills.jsx]: made changes

**Review by ttahm3d — APPROVED** [review]

File: src/utils/cart-checkout.js (hunk 3 above)
Comment (aishwarya-bm) [reply src/utils/cart-checkout.js:14]: Size is controlled in cartreducer and it's required to only change by 1 for add,increment,decrement so I haven't written array reducer for this. For remove item in cart i have used though but it's in the cart reducer action itself

**Pull request merged**